Added Dark/Light Toggle

SubhamKarmakar24 opened the pull request:
Closes #17 

## Changes made:
- Addition of Dark mode toggle button
- The preferred theme that the user has chosen is saved in cache. So, the next time the user opens the website, the preferred theme is already applied.

SubhamKarmakar24 commented:
> 
> 
> Closes #17
> ## Changes made:
> 
>     * Addition of Dark mode toggle button
> 
>     * The preferred theme that the user has chosen is saved in cache. So, the next time the user opens the website, the preferred theme is already applied.

@mihir0699 I have implemented the Dark mode toggle button. Could you please review the code?

mihir0699 commented:
Hello @SubhamKarmakar24 , thanks a lot for the contribution, adding theme toggle is a great feature, your PR was deployed using vercel here- https://video-mihir-git-fork-subhamkarmakar24-main-mihir0699.vercel.app, I have reviewed the changes and great job is done by you, however I would like you to fix some CSS issue before we can go ahead and merge it 
![image](https://user-images.githubusercontent.com/48746544/135707323-f11fed98-f608-4ad5-83ca-ad22360238e9.png)

Can you notice the contrast issues here?

Also in light mode the background that was used to before is not in place, please fix that

SubhamKarmakar24 commented:
> 
> 
> Hello @SubhamKarmakar24 , thanks a lot for the contribution, adding theme toggle is a great feature, your PR was deployed using vercel here- https://video-mihir-git-fork-subhamkarmakar24-main-mihir0699.vercel.app, I have reviewed the changes and great job is done by you, however I would like you to fix some CSS issue before we can go ahead and merge it ![image](https://user-images.githubusercontent.com/48746544/135707323-f11fed98-f608-4ad5-83ca-ad22360238e9.png)
> 
> Can you notice the issues here?
> 
> Also in light mode the background that was used to before is not in place, please fix that

Yes, I would fix them. Regarding the input box, it seems that the library that is used(Ant Design) do not support changing the color of Modal, Input Box and Buttons dynamically. I read their documentation and I could not find a set style attribute that changes the colors. So there is an option of completely changing the library or leave the modal and input boxes as it is now.

The other changes(such as the text and the background color) will be fixed in a moment.

mihir0699 commented:
> > Hello @SubhamKarmakar24 , thanks a lot for the contribution, adding theme toggle is a great feature, your PR was deployed using vercel here- https://video-mihir-git-fork-subhamkarmakar24-main-mihir0699.vercel.app, I have reviewed the changes and great job is done by you, however I would like you to fix some CSS issue before we can go ahead and merge it ![image](https://user-images.githubusercontent.com/48746544/135707323-f11fed98-f608-4ad5-83ca-ad22360238e9.png)
> > Can you notice the issues here?
> > Also in light mode the background that was used to before is not in place, please fix that
> 
> Yes, I would fix them. Regarding the input box, it seems that the library that is used(Ant Design) do not support changing the color of Modal, Input Box and Buttons dynamically. I read their documentation and I could not find a set style attribute that changes the colors. So there is an option of completely changing the library or leave the modal and input boxes as it is now.
> 
> The other changes(such as the text and the background color) will be fixed in a moment.

Okay, do let me know when you are done with the changes, will review it

SubhamKarmakar24 commented:
> Okay, do let me know when you are done with the changes, will review it

@mihir0699 
I have fixed the text color in dark mode, fixed the background color in light mode to original.

mihir0699 commented:
> > Okay, do let me know when you are done with the changes, will review it
> 
> @mihir0699 I have fixed the text color in dark mode, fixed the background color in light mode to original.

Thanks for making the changes, but can you please find a workaround for the input field and button color in dark mode, it does not stands good with the dark mode, rest all looks good now

SubhamKarmakar24 commented:
> 
> 
> > > Okay, do let me know when you are done with the changes, will review it
> > 
> > 
> > @mihir0699 I have fixed the text color in dark mode, fixed the background color in light mode to original.
> 
> Thanks for making the changes, but can you please find a workaround for the input field and button color in dark mode, it does not stands good with the dark mode, rest all looks good now

Okay, I am trying to use pure HTML, CSS components to make the input boxes. Hope it will work

SubhamKarmakar24 commented:
> Thanks for making the changes, but can you please find a workaround for the input field and button color in dark mode, it does not stands good with the dark mode, rest all looks good now

@mihir0699 
Fixed the input background color

mihir0699 commented:
> > Thanks for making the changes, but can you please find a workaround for the input field and button color in dark mode, it does not stands good with the dark mode, rest all looks good now
> 
> @mihir0699 Fixed the input background color

Will review it soon

SubhamKarmakar24 commented:
> Will review it soon

Thanks

mihir0699 commented:
Hello @SubhamKarmakar24 , I have reviewed the changes, the input field issue is also fixed well done, but the chat window colors are not in sync with the theme, can you pls take a look and do the required changes?

SubhamKarmakar24 commented:
> 
> 
> Hello @SubhamKarmakar24 , I have reviewed the changes, the input field issue is also fixed well done, but the chat window colors are not in sync with the theme, can you pls take a look and do the required changes?

Sure, will look into it

mihir0699 commented:
Hello @SubhamKarmakar24 , I hope you are working on this right?

SubhamKarmakar24 commented:
> 
> 
> Hello @SubhamKarmakar24 , I hope you are working on this right?

I tried, but the animations of opening and closing the modal are difficult to replicate.
